pkg/readiness/object_tracker.go

```diff
@@ -99,11 +108,11 @@ func (t *objectTracker) Expect(o runtime.Object) {
 	if _, ok := t.seen[k]; ok {
 		delete(t.seen, k)
 		delete(t.expect, k)
-		t.satisfied[k] = struct{}{}
+		t.satisfied[k] = mutatedObjData(t.mutators...)
```

Comment:
We probably want to leave this as an empty struct to avoid unnecessary memory usage.

Reply:
See my comment in other version of this comment.